rust/operator-binary/src/container.rs

```diff
@@ -958,38 +948,28 @@ wait_for_termination $!
         initial_delay_seconds: i32,
         failure_threshold: i32,
     ) -> Option<Probe> {
-        match self {
-            ContainerConfig::Hdfs {
-                web_ui_http_port_name,
-                web_ui_https_port_name,
-                web_ui_path,
-                ..
-            } => {
-                let http_get_action = if hdfs.has_https_enabled() {
-                    HTTPGetAction {
-                        port: IntOrString::String(web_ui_https_port_name.to_string()),
-                        scheme: Some("HTTPS".into()),
-                        path: Some(web_ui_path.to_string()),
-                        ..HTTPGetAction::default()
-                    }
-                } else {
-                    HTTPGetAction {
-                        port: IntOrString::String(web_ui_http_port_name.to_string()),
-                        scheme: Some("HTTP".into()),
-                        path: Some(web_ui_path.to_string()),
-                        ..HTTPGetAction::default()
-                    }
-                };
-                Some(Probe {
-                    http_get: Some(http_get_action),
-                    period_seconds: Some(period_seconds),
-                    initial_delay_seconds: Some(initial_delay_seconds),
-                    failure_threshold: Some(failure_threshold),
-                    ..Probe::default()
-                })
-            }
-            _ => None,
-        }
+        let ContainerConfig::Hdfs {
+            web_ui_http_port_name,
+            web_ui_https_port_name,
+            ..
+        } = self
+        else {
+            return None;
+        };
+
+        let port = if hdfs.has_https_enabled() {
+            web_ui_https_port_name
+        } else {
+            web_ui_http_port_name
+        };
+
+        Some(Probe {
+            tcp_socket: Some(Self::tcp_socket_action_for_port(*port)),
```

Comment:
```suggestion
            // Use tcp_socket instead of http_get so that the probe is independent of the authentication settings.
            tcp_socket: Some(Self::tcp_socket_action_for_port(*port)),
```

or something similar